runtime/domain.c

```diff
@@ -1462,10 +1473,23 @@ int caml_try_run_on_all_domains(
                                                  leader_setup, 0, 0);
 }
 
-void caml_interrupt_self(void) {
+void caml_interrupt_self(void)
+{
   interrupt_domain(&domain_self->interruptor);
 }
 
+/* async-signal-safe */
+void caml_interrupt_all_for_signal(void)
+{
+  for (dom_internal *d = all_domains; d < &all_domains[Max_domains]; d++) {
+    atomic_uintnat * interrupt_word =
+      atomic_load_explicit(&d->interruptor.interrupt_word,
+                           memory_order_acquire);
+    if (interrupt_word == NULL) return;
```

Comment:
Instead of looping over `all_domains`, wouldn't it be simpler to just check for every domains that are part of the STW participating set?
This may be a naive suggestion, there may be some odd cases where it is important to check for `interrupt_word` nonetheless, but to me checking the participating set solves both having to check for `interrupt_word == NULL` and doing a for loop over `all_domains`.